net: lwm2m: fix time serialized records for resource instances

walzsi opened the pull request:
This PR fixes the issue by ensuring that a name entry is generated for every time-serialized record, including resource instances.

Also two new test cases have been added to the SenML CBOR test to verify correct handling of time-serialized records.

walzsi committed:
net: lwm2m: fix missing name entries for time-serialized resource instances

The LwM2M SENML JSON/CBOR decoder expects a name entry for every
time-serialized record. While this was already the case for resources,
resource instances were missing their corresponding name entries. This
change ensures consistency by adding name entries for resource instances
as well.

Signed-off-by: Simon Walz <simon.walz@autosen.com>

rlubos reviewed (commented):
Please fix the compliance nits.

Also, `CONFIG_POSIX_CLOCK` was deprecated quite some time ago and is now removed, not sure why would it be needed here though?

walzsi committed:
tests: lwm2m: add test for senml cbor composite writer

This adds a test for the composite writer with time serialized resource and
resource instance values.

Signed-off-by: Simon Walz <simon.walz@autosen.com>
